### ISSUE TITLE
[All generators] Cannot generate when `allOf` property is in /paths

### ISSUE THREAD

**Issue by tagliala** [swagger-api/swagger-codegen#6519]
##### Description

Cannot generate documentation when the `allOf` property is contained in a `schema` inside `/paths`

##### Swagger-codegen version

2.2.3

##### Swagger declaration file content or url

```yaml
swagger: "2.0"
info:
  description: "allOf issue demonstration"
  version: "1.0.0"
  title: "Swagger allOf regression"
host: "example.com"
schemes:
- "http"
paths:
  /rides:
    post:
      tags:
      - "rides"
      summary: "Add a new ride"
      description: ""
      operationId: "addRide"
      parameters:
      - in: "body"
        name: "body"
        description: "Ride object"
        required: true
        schema:
          allOf:
            - $ref: "#/definitions/Ride"
      responses:
        405:
          description: "Invalid input"
definitions:
  Ride:
    type: "object"
    required:
    - "from"
    - "to"
    properties:
      id:
        type: "integer"
        format: "int64"
      from:
        type: "string"
        example: "London"
      to:
        type: "string"
        example: "Manchester"
```

##### Command line used for generation

```sh
$ swagger-codegen generate -i swagger.yaml  -l html2
```

##### Steps to reproduce

Run the above command line with the provided configuration file

```
[main] INFO io.swagger.parser.Swagger20Parser - reading from swagger.yaml
Exception in thread "main" java.lang.RuntimeException: Could not process operation:
  Tag: rides
  Operation: addRide
  Resource: post /rides
  Definitions: {Ride=io.swagger.models.ModelImpl@c32f7a5d}
  Exception: null
	at io.swagger.codegen.DefaultGenerator.processOperation(DefaultGenerator.java:861)
	at io.swagger.codegen.DefaultGenerator.processPaths(DefaultGenerator.java:764)
	at io.swagger.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:388)
	at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:700)
	at io.swagger.codegen.cmd.Generate.run(Generate.java:285)
	at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:35)
Caused by: java.lang.NullPointerException
	at io.swagger.codegen.DefaultCodegen.isDataTypeBinary(DefaultCodegen.java:2682)
	at io.swagger.codegen.DefaultCodegen.fromParameter(DefaultCodegen.java:2633)
	at io.swagger.codegen.DefaultCodegen.fromOperation(DefaultCodegen.java:2191)
	at io.swagger.codegen.languages.StaticHtml2Generator.fromOperation(StaticHtml2Generator.java:158)
	at io.swagger.codegen.DefaultGenerator.processOperation(DefaultGenerator.java:808)
	... 5 more
```

##### Related issues/PRs

swagger-api/swagger-editor#1497

##### Suggest a fix/enhancement

Please take a look at swagger-api/swagger-editor#1497, there is plenty of information.

Problem is in the `allOf` property

Lines generating the issue are:
```yml
          schema:
            allOf:
              - $ref: '#/definitions/Ride'
```

If you remove `allOf:`, as in:

```yml
          schema:
            $ref: '#/definitions/Ride'
```

documentation would be generated. This was working with swagger editor 3.1.6, but reference resolution changed and now it doesn't work anymore

**Comment by webron** [comment]
@wing328 any thoughts on this?

**Comment by wing328** [comment]
@tagliala @webron I could repeat the issue with the latest master and agree it shouldn't throw NPE. I'll look into this issue tomorrow to see if I can come up with a PR to better handle the NPE.

**Comment by wing328** [comment]
@tagliala I've fixed the issue via #6635 

Please pull the latest master to give it a try or use the SNAPSHOT version as mentioned in the README.

**Comment by tagliala** [comment]
Hi @wing328 

thanks for your fix. At the moment I'm not able to test against the master branch, I will wait for a new release so I can test through the online editor

**Comment by tagliala** [comment]
Hi, there has been a regression in version 2.4.0 and I'm not able to generate the docs anymore

The file is the very same

2.4.0+ (included 3.0.3)

```sh
$ /usr/local/opt/swagger-codegen@2/bin/swagger-codegen version
2.4.0
$ /usr/local/opt/swagger-codegen@2/bin/swagger-codegen generate -i regressionallof.yml -l html2
[main] INFO io.swagger.parser.Swagger20Parser - reading from regressionallof.yml
[main] INFO io.swagger.codegen.ignore.CodegenIgnoreProcessor - No .swagger-codegen-ignore file found.
Exception in thread "main" java.lang.RuntimeException: Could not process operation:
  Tag: Tag {
	name: rides
	description: null
	externalDocs: null
	extensions:{}}
  Operation: addRide
  Resource: post /rides
  Definitions: {Ride=io.swagger.models.ModelImpl@337ac83}
  Exception: null
	at io.swagger.codegen.DefaultGenerator.processOperation(DefaultGenerator.java:940)
	at io.swagger.codegen.DefaultGenerator.processPaths(DefaultGenerator.java:819)
	at io.swagger.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:439)
	at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:754)
	at io.swagger.codegen.cmd.Generate.run(Generate.java:285)
	at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:35)
Caused by: java.lang.NullPointerException
	at io.swagger.codegen.DefaultCodegen.getExamples(DefaultCodegen.java:3828)
	at io.swagger.codegen.DefaultCodegen.fromOperation(DefaultCodegen.java:2275)
	at io.swagger.codegen.languages.StaticHtml2Generator.fromOperation(StaticHtml2Generator.java:158)
	at io.swagger.codegen.DefaultGenerator.processOperation(DefaultGenerator.java:889)
	... 5 more
```

2.3.1
```
$ java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar version
2.3.1
$ java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -i regressionallof.yml -l html2 -o test
[main] INFO io.swagger.parser.Swagger20Parser - reading from regressionallof.yml
[main] INFO io.swagger.codegen.AbstractGenerator - writing file ./test/index.html
[main] INFO io.swagger.codegen.AbstractGenerator - writing file ./test/.swagger-codegen/VERSION
```

Should I open a new issue?

**Comment by HugoMario** [comment]
hey @tagliala i just added a PR (#9146) to fix the regresion. can you please check when you have a chance and let me know?
